hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java

```diff
@@ -549,21 +567,48 @@ public RecordReader getRecordReader(InputSplit split, JobConf job, Reporter repo
       throw new IOException("cannot find class " + inputFormatClassName);
     }
 
-    pushProjectionsAndFilters(job, inputFormatClass, hsplit.getPath(0));
-
+    invokePushProjectAndFilters(job, inputFormatClass, hsplit.getPath(0));
     if (inputFormatClass.getName().equals(getParquetRealtimeInputFormatClassName())) {
       HoodieCombineFileInputFormatShim shims = createInputFormatShim();
       IOContextMap.get(job).setInputPath(((CombineHiveInputSplit) split).getPath(0));
       return shims.getRecordReader(job, ((CombineHiveInputSplit) split).getInputSplitShim(),
           reporter, CombineHiveRecordReader.class);
     } else {
-      return ShimLoader.getHadoopShims().getCombineFileInputFormat().getRecordReader(job, (CombineFileSplit) split,
-          reporter, CombineHiveRecordReader.class);
+      return ShimLoader.getHadoopShims().getCombineFileInputFormat()
+          .getRecordReader(job, (CombineFileSplit) split,
+              reporter, CombineHiveRecordReader.class);
+    }
+  }
+
+  private void invokePushProjectAndFilters(JobConf job, Class<?> inputFormatClass, Path splitPath)
+      throws IOException {
```

Comment:
@danny0405 Is it will be better if we add a shim class to do `pushProjectionsAndFilters`?